### ISSUE TITLE
flake: Extended.deploymentconfigs with minimum ready seconds set [Conformance] should not transition the deployment to Complete before satisfied

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#11544]
https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/10573/

https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/7766/testReport/junit/(root)/Extended/deploymentconfigs_with_minimum_ready_seconds_set__Conformance__should_not_transition_the_deployment_to_Complete_before_satisfied/

```
Extended.deploymentconfigs with minimum ready seconds set [Conformance] should not transition the deployment to Complete before satisfied (from (conformance_parallel_01.xml))

Failing for the past 1 build (Since Failed#7766 )
Took 5 min 4 sec.
add description
Stacktrace

/data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:134
Oct 24 16:13:13.630: Couldn't delete ns: "extended-test-cli-deployment-oxfku-nl315": namespace extended-test-cli-deployment-oxfku-nl315 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed (&errors.errorString{s:"namespace extended-test-cli-deployment-oxfku-nl315 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed"})
/data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:338
```

**Comment by smarterclayton** [comment]
Saw this today, have not seen it before

**Comment by smarterclayton** [comment]
```
------------------------------
deploymentconfigs with minimum ready seconds set [Conformance] 
  should not transition the deployment to Complete before satisfied
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:783

[BeforeEach] [Top Level]
  /data/src/github.com/openshift/origin/test/extended/util/test.go:45
[BeforeEach] deploymentconfigs
  /data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:133
STEP: Creating a kubernetes client
Oct 24 16:08:09.486: INFO: >>> kubeConfig: /tmp/openshift/test-extended/core/openshift.local.config/master/admin.kubeconfig

STEP: Building a namespace api object
Oct 24 16:08:09.510: INFO: configPath is now "/tmp/extended-test-cli-deployment-oxfku-nl315-user.kubeconfig"
Oct 24 16:08:09.510: INFO: The user is now "extended-test-cli-deployment-oxfku-nl315-user"
Oct 24 16:08:09.510: INFO: Creating project "extended-test-cli-deployment-oxfku-nl315"
STEP: Waiting for a default service account to be provisioned in namespace
[It] should not transition the deployment to Complete before satisfied
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:783
Oct 24 16:08:09.621: INFO: Running 'oc create --config=/tmp/extended-test-cli-deployment-oxfku-nl315-user.kubeconfig --namespace=extended-test-cli-deployment-oxfku-nl315 -f /data/src/github.com/openshift/origin/test/extended/testdata/deployments/deployment-min-ready-seconds.yaml -o name'
STEP: verifying the deployment is marked running
STEP: verifying that all pods are ready
STEP: verifying that the deployment is still running
[AfterEach] with minimum ready seconds set [Conformance]
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:739
[AfterEach] deploymentconfigs
  /data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:134
Oct 24 16:08:13.516: INFO: Waiting up to 1m0s for all nodes to be ready
STEP: Destroying namespace "extended-test-cli-deployment-oxfku-nl315" for this suite.
Oct 24 16:13:13.612: INFO: namespace: extended-test-cli-deployment-oxfku-nl315, resource: bindings, ignored listing per whitelist
Oct 24 16:13:13.630: INFO: Couldn't delete ns: "extended-test-cli-deployment-oxfku-nl315": namespace extended-test-cli-deployment-oxfku-nl315 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed (&errors.errorString{s:"namespace extended-test-cli-deployment-oxfku-nl315 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed"})


• Failure in Spec Teardown (AfterEach) [304.145 seconds]
deploymentconfigs
/data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:785
  with minimum ready seconds set [Conformance] [AfterEach]
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:784
    should not transition the deployment to Complete before satisfied
    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:783

    Oct 24 16:13:13.630: Couldn't delete ns: "extended-test-cli-deployment-oxfku-nl315": namespace extended-test-cli-deployment-oxfku-nl315 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed (&errors.errorString{s:"namespace extended-test-cli-deployment-oxfku-nl315 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed"})

    /data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:338
------------------------------
```

**Comment by kargakis** [comment]
I have seen ns deletion before flaking for another [deployment test](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/7195/testReport/junit/(root)/Extended/deploymentconfigs_when_run_iteratively__Conformance__should_immediately_start_a_new_deployment/).
```
Extended.deploymentconfigs when run iteratively [Conformance] should immediately start a new deployment (from (conformance_parallel_01.xml)) 

/data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:134
Oct 13 06:18:13.310: Couldn't delete ns: "extended-test-cli-deployment-es6td-4w890": namespace extended-test-cli-deployment-es6td-4w890 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed (&errors.errorString{s:"namespace extended-test-cli-deployment-es6td-4w890 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed"})
/data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:338
```
This is similar to the limit range flake: https://github.com/openshift/origin/issues/11094

**Comment by kargakis** [comment]
Should be fixed by #11627
